src/persist-client/src/internal/state_serde.json

```diff
@@ -85,121 +85,303 @@
     }
   },
   "since": [
-    16259358130896200968
+    18149886410317824892
   ],
   "upper": [
-    9048289724302440207
+    16874249011475307565
   ],
   "batches": [
     {
       "lower": [
         0
       ],
       "upper": [
-        8416425234532291380
+        9048289724302440207
       ],
       "since": [
-        76773281222928414
+        16259358130896200968
       ],
-      "len": 5,
-      "part_runs": []
+      "len": 7,
+      "part_runs": [
+        [
+          {
+            "key": "𝕏`𑜈Ⱥ5:<",
+            "encoded_size_bytes": 16872256722443379164,
+            "stats": {
+              "len": 14178690985928908946,
+              "cols": {
+                "<?Ѩ𑤉Gr𞺇&ᜄѨcѨ𑴣<෮:`�&ῳ/o": {
+                  "len": 16114378244580388529
+                },
+                "{ꪵ¥೦᧞ì": {
+                  "lower": -6248419378273962495,
+                  "upper": 4066393799215850741
+                }
+              }
+            }
+          }
+        ],
+        []
+      ]
     },
     {
       "lower": [
-        8416425234532291380
+        9048289724302440207
       ],
       "upper": [
-        8852211157054239750
+        10179017796861612496
       ],
       "since": [
-        2551556528472692109
+        18149886410317824892
       ],
-      "len": 5,
+      "len": 6,
       "part_runs": [
         [
           {
-            "key": "=𖭕`<?Ѩ𑤉Gr𞺇&ᜄѨ",
-            "encoded_size_bytes": 16884651883333590719,
+            "key": "᪭^d𝪡\\𑁧¼🢤s𑲡𑌁𝔚",
+            "encoded_size_bytes": 7409370903940415159,
             "stats": {
-              "len": 1545472491260012506,
-              "cols": {}
+              "len": 2825276381931333665,
+              "cols": {
+                "<Ꚙ+ໆᛌ)🕴/\\$🕴🕴𞺡𐪌ë𑇯𖨂$G್𞺧:$": {
+                  "len": 18042306332385152625
+                },
+                "k{": {
+                  "lower": false,
+                  "upper": true
+                },
+                "𛅐(q*Ο{Ѩ%🕴{ßȺ:": {
+                  "len": 474717531799020000
+                },
+                "𞅎ᄌI﹨ᮠѨ%ü'&E🫧\\E𞺍$𝑙$`Nૈ_*ຉ": {
+                  "len": 11984174003953907109,
+                  ",🪇🉡$=|/e🄔Dm<~`ౡ🕴Ფ9.": {
+                    "len": 18300643318019195797
+                  },
+                  "=Ἔ": {
+                    "lower": "[75, 7f, df, 83, 4e, a7, 8, b, b1, 4c, b6]",
```

Comment:
Feels like this would be easier to use programmatically as either a "dense" hex string (`"757fdf..."`) or an array of integers: (`[0x75, 0x7f, 0xdf, ...]`, though JSON would write those out as decimal). But if it's annoying that doesn't seem particularly important...

Reply:
fair. i table flipped when i found out that you need an external crate to print a Vec<u8> as a hex string, but we've already got it in our dep graph, so may as well. done!